exotic.py

```diff
@@ -1590,163 +1711,177 @@ def parse_args():
-                    firstimagecounter = firstimagecounter + 1
-                    firstImageData = sortedallImageData[firstimagecounter]
-
-            # Filter the other data as well
-            sortedallImageData = sortedallImageData[firstimagecounter:]
-            timesListed = timesListed[firstimagecounter:]
-            airMassList = airMassList[firstimagecounter:]
-            sortedTimeList = sortedTimeList[firstimagecounter:]
-
-            # apply cals correction if applicable
-            if darksBool:
-                print("Dark subtracting images.")
-                sortedallImageData = sortedallImageData - generalDark
-            elif biasesBool:
-                print("Bias-correcting images.")
-                sortedallImageData = sortedallImageData - generalBias
-            else:
-                pass
-
-            if flatsBool:
-                print("Flattening images.")
-                sortedallImageData = sortedallImageData / generalFlat
-
-            pathSolve = saveDirectory + 'first_file.fits'
-            try:
-                os.remove(pathSolve)
-            except OSError:
-                pass
-            convertToFITS = fits.PrimaryHDU(data=sortedallImageData[0])
-            convertToFITS.writeto(pathSolve)
-            wcsFile = check_wcs(pathSolve, saveDirectory)
-            if wcsFile:
-                hdulWCS = fits.open(name=wcsFile, memmap=False, cache=False, lazy_load_hdus=False)  # opens the fits file
-                rafile, decfile = get_radec(hdulWCS)
-
-            print("\nAligning your images from .FITS. Please wait.")
-            done = False
-            t = threading.Thread(target=animate, daemon=True)
-            t.start()
-            sortedallImageData, unalignedBoolList, boollist = image_alignment(sortedallImageData)
-            done = True
-            print('\n\nImages Aligned.')
+                    os.remove(pathSolve)
+                except OSError:
+                    pass
+                convertToFITS = fits.PrimaryHDU(data=sortedallImageData[0])
+                convertToFITS.writeto(pathSolve)
+                wcsFile = check_wcs(pathSolve, saveDirectory)
+
+                # Check pixel coordinates by converting to WCS. If not correct, loop over again
+                if wcsFile:
+                    hdulWCS = fits.open(name=wcsFile, memmap=False, cache=False, lazy_load_hdus=False)  # opens the fits file
+
+                    # Save previously entered x and y pixel coordinates before checking against plate solution
+                    saveUIprevTPX, saveUIprevTPY = UIprevTPX, UIprevTPY
+                    UIprevTPX, UIprevTPY = check_targetpixelwcs(UIprevTPX, UIprevTPY, pDict['ra'],
+                                                                pDict['dec'], hdulWCS)
+
+                    # If the coordinates were not changed, do not loop over again
+                    if UIprevTPX != saveUIprevTPX or UIprevTPY != saveUIprevTPY:
+                        continue
```

Comment:
If either target coordinate changes, it will loop over again (at most 2 times).

Reply:
Why 2 times?

Reply:
If the user entered the target coordinates correct the first time, it will loop once. If the user entered the target coordinates incorrectly, it will loop over again once it hits the check_targetpixelcoords() function and have the user enter in the correct ones. Once that happens, the time sorting and plate solution will be redone.